curator-recipes/src/test/java/org/apache/curator/framework/recipes/nodes/TestPersistentNode.java

```diff
@@ -136,4 +138,52 @@ public void testQuickCloseNodeExists() throws Exception
             CloseableUtils.closeQuietly(client);
         }
     }
+
+    @Test
+    public void testInterruption() throws Exception
```

Comment:
This issue is a root cause of some significant problems we've encountered. I'll have another look at this and come up with a more pertinent test case.